core/src/main/java/org/mule/runtime/core/execution/ModuleFlowProcessingPhase.java

```diff
@@ -159,6 +183,35 @@ public void runPhase(final ModuleFlowProcessingPhaseTemplate template, final Mes
     }
   }
 
+  private Event createEvent(ModuleFlowProcessingPhaseTemplate template, MessageProcessContext messageProcessContext,
+                            ComponentIdentifier sourceIdentifier)
+      throws MuleException {
+    Message message = template.getMessage();
+    Event templateEvent =
```

Comment:
there seems to be no need to create this event at the beginning. Just a builder would do